docs/gsoc2020_ideas.rst

```diff
@@ -0,0 +1,134 @@
+
+We expect from students that they're at least comfortable with Python
+(intermediate level). Some projects may also require C++ or C skills.
+Knowing how to use Git is also important; this can be learned before the
+official start of GSoC if needed though.
+
+If you have an idea of what you would like to work on (see below for ideas)
+and are considering participating:
+
+
+1. Read the `PSF page <http://python-gsoc.org/>`_ carefully, it contains
+   important advice on the process.
+2. Read `advice on writing a proposal <http://turnbull.sk.tsukuba.ac.jp/Blog/SPAM.txt#how-to-spam-in-detail>`_
+   (written with the Mailman project in mind, but generally applicable)
+3. Look at the guidelines on how to contribute to Scipy.
+4. Make a enhancement/bugfix/documentation fix -- it does not have to be big,
+   and it does not need to be related to your proposal. Doing so before
+   applying for the GSoC is a hard requirement for ``uarray``. It helps
+   everyone you get some idea how things would work during GSoC.
+5. Start writing your proposal early, post a draft to the the mentors' email
+   addresses mailing list and iterate based on the feedback you receive. This
+   will both improve the quality of your proposal and help you find a suitable mentor.
+
+Contact
+-------
+
+If you have a question *after checking all guideline pages above*, you can
+email the mentors.
+
+``uarray`` project ideas
+------------------------
+
+``uarray``: Add querying for state
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Adding querying for the ``uarray._BackendState`` object will allow users of
+``uarray`` to see what's inside the opaque object. Some parts can be re-used
+from the pickling machinery.
+
+It can also help downstream users to access the parameters of the currently
+set backend, which is a planned feature of ``uarray``. Here is a list of goals
+for this project:
+
+* Allow downstream projects to query the list of backends.
+* Allow downstream projects to query the list of parameters for a backend.
+
+This would enable, for example, the following use-cases:
+
+* Allow a downstream library to detect a backend and run specialised code for
+  it.
+* Allow a downstream library to fail-fast on a known-unsupported backend.
+
+This project has a straightforward design and needs some implementation work,
+and will require interacting with the mentors to implement and polish.
+
+* Required knowledge: Python C-API and C++
+* Difficulty level: easy
+* Potential mentors: Peter Bell and Hameer Abbasi
+
+``uarray``: Allow subdomains
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This idea would allow a backend to encompass functions from more than one domain.
+
+The primary goal of this project would be:
+
+* Develop a system that allows, via some kind of matching mechanism, to select
+  which domains it supports, **while maintaining backward compatibility**.
+
+This would allow a backend targeting NumPy to also target, for example, the
+``numpy.random`` submodule.
```

Comment:
I'm not sure this clarifies it. If you're targeting (all of) NumPy, you're already targeting `numpy.random` as well?

This whole idea needs a bit more detail, it's unclear to me from a single read what the work would entail.

Reply:
Is the text I added enough?

Reply:
Yes, clear now - thanks